### PR TITLE
expression: implement vectorized evaluation for builtinArithmeticIntDivideIntSig

### PR DESCRIPTION
**Pull request by silver--bullet**
PCP #12103

### What problem does this PR solve? 
expression: implement vectorized evaluation for builtinArithmeticIntDivideIntSig

### What is changed and how it works?
It gets faster about 150%~260%

### Check List 

Tests 

 - Benchmark
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinArithmeticFunc -run=BenchmarkVectorizedBuiltinArithmeticFunc -args builtinArithmeticIntDivideIntSig
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-VecBuiltinFunc-12                86403             13099 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-NonVecBuiltinFunc-12             34476             34652 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-VecBuiltinFunc#01-12            146065              8312 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-NonVecBuiltinFunc#01-12          39729             30763 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-VecBuiltinFunc#02-12            122324              9855 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-NonVecBuiltinFunc#02-12          34365             33425 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-VecBuiltinFunc#03-12            104485             10976 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinArithmeticFunc/builtinArithmeticIntDivideIntSig-NonVecBuiltinFunc#03-12          37557             31632 ns/op               0 B/op          0 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      11.269s

```

### CODE REVIEW

File: expression/builtin_arithmetic_vec.go [hunk 1]
```diff
@@ -734,11 +734,117 @@ func (b *builtinArithmeticDivideRealSig) vecEvalReal(input *chunk.Chunk, result
...
+func (b *builtinArithmeticIntDivideIntSig) divideUU(result *chunk.Column, lhi64s, rhi64s, resulti64s []int64) error {
+	for i := 0; i < len(lhi64s); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		lh, rh := lhi64s[i], rhi64s[i]
+		if rh == 0 {
+			result.SetNull(i, true)
+			return handleDivisionByZeroError(b.ctx)
+		}
```
Comment (wshwsh12) [inline expression/builtin_arithmetic_vec.go]: The check order is different from non-vec function's. 
1. Check arg1 is null.
2. Check arg1 == 0 .
3. Check arg0 is null.

File: expression/builtin_arithmetic_vec.go [hunk 2]
```diff
@@ -734,11 +734,117 @@ func (b *builtinArithmeticDivideRealSig) vecEvalReal(input *chunk.Chunk, result
...
+
+func (b *builtinArithmeticIntDivideIntSig) divideUU(result *chunk.Column, lhi64s, rhi64s, resulti64s []int64) error {
+	for i := 0; i < len(lhi64s); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		lh, rh := lhi64s[i], rhi64s[i]
+		if rh == 0 {
+			result.SetNull(i, true)
+			return handleDivisionByZeroError(b.ctx)
```
Comment (wshwsh12) [inline expression/builtin_arithmetic_vec.go]: Handle the error and continue, not return.

File: expression/builtin_arithmetic_vec.go [hunk 3]
```diff
@@ -734,11 +734,117 @@ func (b *builtinArithmeticDivideRealSig) vecEvalReal(input *chunk.Chunk, result
...
+		}
+		val, err := types.DivUintWithInt(uint64(lh), rh)
```
Comment (wshwsh12) [inline expression/builtin_arithmetic_vec.go]: Now we use the arg0 to calculate result before checking arg0.isNull(i). If arg0.isNull(i) is true, arg0[i]  is invalid and we don't know what value the arg0[i] is.  It maybe get an error.

File: expression/builtin_arithmetic_vec.go [hunk 4]
```diff
@@ -734,11 +734,117 @@ func (b *builtinArithmeticDivideRealSig) vecEvalReal(input *chunk.Chunk, result
...
+	case isLHSUnsigned && isRHSUnsigned:
+		err = b.divideUU(result, lhi64s, rhi64s, resulti64s)
+	case isLHSUnsigned && !isRHSUnsigned:
+		err = b.divideUS(result, lhi64s, rhi64s, resulti64s)
+	case !isLHSUnsigned && isRHSUnsigned:
+		err = b.divideSU(result, lhi64s, rhi64s, resulti64s)
+	case !isLHSUnsigned && !isRHSUnsigned:
+		err = b.divideSS(result, lhi64s, rhi64s, resulti64s)
+	}
+	result.MergeNulls(lh)
```
Comment (qw4990) [inline expression/builtin_arithmetic_vec.go]: Please move this statement to `line:763`

**Pull request closed without merging**